===================================================================
RCS file: /var/cvsup/gsl/vcl/source/gdi/bitmap.cxx,v
retrieving revision 1.7.178.1
retrieving revision 1.7.178.2
diff -u -r1.7.178.1 -r1.7.178.2
--- vcl/source/gdi/bitmap.cxx	2003/11/25 18:08:00	1.7.178.1
+++ vcl/source/gdi/bitmap.cxx	2003/12/10 18:25:44	1.7.178.2
@@ -1792,7 +1792,7 @@
     if (eColorMode == BMP_COLOR_HIGHCONTRAST && GetBitCount() >= 24) {
 
 	    // Yet another special case for icons
-	    BitmapReadAccess* pAcc = AcquireReadAccess();
+	    BitmapReadAccess* pAcc = ((Bitmap*)this)->AcquireReadAccess();
 	    BitmapWriteAccess* pDis = aRet.AcquireWriteAccess();
 
 	    Size aSize = GetSizePixel();
@@ -1811,7 +1811,7 @@
 	    }
 
 	    aRet.ReleaseAccess( pDis );
-	    ReleaseAccess( pAcc );
+	    ((Bitmap*)this)->ReleaseAccess( pAcc );
     } else {
 	
 		Color*  pSrcColors = NULL;
===================================================================
RCS file: /var/cvsup/gsl/vcl/source/gdi/bitmap2.cxx,v
retrieving revision 1.4.178.1
retrieving revision 1.4.178.2
diff -u -r1.4.178.1 -r1.4.178.2
--- bitmap2.cxx	2003/11/25 18:08:00	1.4.178.1
+++ vcl/source/gdi/bitmap2.cxx	2003/12/10 18:25:45	1.4.178.2
@@ -1079,7 +1079,7 @@
 
 	aAlpha.Erase( COL_WHITE );
 
-	BitmapReadAccess *pSrc = AcquireReadAccess();
+	BitmapReadAccess *pSrc = ((Bitmap*)this)->AcquireReadAccess();
 	BitmapWriteAccess *pDst = aAlpha.AcquireWriteAccess();
 
 	for( long nY = 0; nY < aSize.Height(); nY++ )
@@ -1092,8 +1092,8 @@
 		}
 	}
 	
-	ReleaseAccess( pSrc );
-	ReleaseAccess( pDst );
+	((Bitmap*)this)->ReleaseAccess( pSrc );
+	((Bitmap*)this)->ReleaseAccess( pDst );
 		
 	return aAlpha;
 }
===================================================================
RCS file: /var/cvsup/gsl/vcl/source/gdi/image.cxx,v
retrieving revision 1.8.202.1
retrieving revision 1.8.202.2
diff -u -r1.8.202.1 -r1.8.202.2
--- image.cxx	2003/11/25 18:08:02	1.8.202.1
+++ vcl/source/gdi/image.cxx	2003/12/10 18:25:47	1.8.202.2
@@ -804,9 +804,11 @@
 		return;
 	}
 
+    BOOL bUseAlphaMask = FALSE;
 	if( !rMaskBmp && rBitmap.GetBitCount() == 32 )
 	{
-		rMaskBmp = rBitmap.GetAlphaMask();
+        bUseAlphaMask = TRUE;
+		//rMaskBmp = rBitmap.GetAlphaMask();
 		bColor = FALSE;
 	}
 
@@ -859,7 +861,7 @@
 	}
 
 	pThis->mpImplData->mpImageBitmap = new ImplImageBmp;
-	pThis->mpImplData->mpImageBitmap->Create( rBitmap, rMaskBmp,
+    pThis->mpImplData->mpImageBitmap->Create( rBitmap, bUseAlphaMask ? rBitmap.GetAlphaMask() : rMaskBmp,
 											  rColor, bColor,
 											  pThis->mpImplData->maImageSize.Width(),
 											  pThis->mpImplData->maImageSize.Height(),
